.github/workflows/release.yml

```diff
@@ -0,0 +1,39 @@
+on:
+  push:
+    branches: main
+
+name: Release
+
+jobs:
+  create_release:
+    runs-on: ubuntu-latest
+    if: startsWith(github.event.head_commit.message, 'release')
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set version from commit message
+        id: tag_id
+        run: |
+          VAR="${{ github.event.head_commit.message }}"
+          TAG="${VAR#* }"
+          echo ::set-output name=tag::"${TAG}"
+          echo "${TAG}"
```

Comment:
It seems that generally recommended to avoid using `github.event.head_commit.message` to refer to a commit message.
https://rhysd.github.io/actionlint/#eJyFk9Fv0zAQxt/zV5wK0hgiKQiJB0uTtiE0ENKQRsceIye5NGGJHdnnFmnsf+dimyxtGX1plbv74t/3naOVSAAGZ5vxH6AwUpUNWgG9bFWSKNmjgBvsUFpMkp+6sONgaVAS5ibUg9Q4ZVOtBLjCKXJpxxOWfKutBViShuxdS82rNf+4IsMNKsoalFVe6r5vKevRWrnGN3AS33xy6vWWcLDhFIAUnB0BZUmtVnbJuOW9dnS+eT9NBOzvSLBBY3kMaqN7CMdAPCZOM14lgOQ6b6upxGYE/J4eAX5c3JwtXj48wDF4eHxczHSri6tRx/IXr2Gng2WjQQiLlDL94MhTnzGIEKxg4eH4Xn1uNCbmKzvOYuO/7p5hiVJ/iPDu/SayEFYWRv3jzPVfqEvXD1P8UlX+lo05H+Qeh6Z62G8vqWnRVY60M8sQe8rjaVj8+eZD9naSbLktZobCeE76HtmqJ0e+tAx79WX1+fYyX337+umaqWeaCmvpOsoLBhdQy87irFs6S7rPx/XAkSRmqrE/GKzbXwIW+wl99F/R9HXt2ldlO+iu08u4hMn2u2dN/4MtRjsBKtzmB5CB5kn4dGMm3ewm7GrXqNCwi2jiWtMIT8bNs5PsY3s7VNI395M9vqQ/E+tnpA==

Wouldn't the process of creating a release when a tag is push be better?
For example, the following well-known extensions would use that method.
https://github.com/prettier/prettier-vscode/blob/b395990a96e1cbd754646de0a2f21c4b36593cd3/.github/workflows/main.yml

Reply:
Fixed in d62736dbc191a6e60d173d7a0fbfc923bfad597d as per https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-an-intermediate-environment-variable

Reply:
The prettier-vscode workflow seems overkill, it installs Node.JS, and many stuff, creates release assets while those are on the marketplace.

Reply:
> Wouldn't the process of creating a release when a tag is push be better?

@eitsupi Maybe ...
Although for now no tags were ever created while commit messages in the form "release ..." were, so I thought this was the more appropriate action without changing @jjallaire workflow to release new VSCode extension version.

Reply:
> Although for now no tags were ever created while commit messages in the form "release ..." were, so I thought this was the more appropriate action without changing @jjallaire workflow to release new VSCode extension version.

Of course, I understand your intentions.
I just wondered if it might be possible to run a tag-based release process similar to the quarto cli.

> The prettier-vscode workflow seems overkill, it installs Node.JS, and many stuff, creates release assets while those are on the marketplace.

I understand that this is not necessary at this time, but I just wanted to provide an example of automating the publication to the marketplace.
In fact, it would have been better if I could have found an example that also runs the publication to Open-VSX.

Reply:
Indeed, I was also thinking of a unified framework to do both releases on GitHub and marketplace, but I am not familiar at all with the latter.